src/Network/Network/VirtualNetworkGateway/StartAzVirtualNetworkGatewayPacketCaptureCommand.cs

```diff
@@ -0,0 +1,119 @@
+        [ValidateNotNullOrEmpty]
+        public PSVirtualNetworkGateway InputObject { get; set; }
+
+        [Parameter(
+            ParameterSetName = "ByResourceId",
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The Azure resource ID of the VirtualNetworkGateway where packet capture to be started.")]
+        [ValidateNotNullOrEmpty]
+        public string ResourceId { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "Filter options for start packet capture on virtual network gateway.")]
+        public string FilterData { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "Run cmdlet in the background")]
+        public SwitchParameter AsJob { get; set; }
+
+        public override void Execute()
+        {
+            base.Execute();
+            PSVirtualNetworkGateway existingVirtualNetworkGateway = null;
+            if (ParameterSetName.Equals("ByInputObject"))
+            {
+                existingVirtualNetworkGateway = this.InputObject;
+                this.ResourceGroupName = this.InputObject.ResourceGroupName;
+                this.Name = this.InputObject.Name;
+            }
+            else
+            {
+                if (ParameterSetName.Equals("ByResourceId"))
+                {
+                    var parsedResourceId = new ResourceIdentifier(ResourceId);
+                    Name = parsedResourceId.ResourceName;
+                    ResourceGroupName = parsedResourceId.ResourceGroupName;
+                }
+
+                existingVirtualNetworkGateway = this.GetVirtualNetworkGateway(this.ResourceGroupName, this.Name);
+            }
+
+            if (existingVirtualNetworkGateway == null)
+            {
+                throw new PSArgumentException(Properties.Resources.ResourceNotFound, "Virtual Network Gateway");
+            }
+
+            VpnPacketCaptureStartParameters parameters = new VpnPacketCaptureStartParameters();
+            if (this.FilterData != null)
+            {
+                parameters.FilterData = FilterData;
+            }
+
+            WriteVerbose(String.Format(Properties.Resources.CreatingLongRunningOperationMessage, this.ResourceGroupName, this.Name));
+            PSVirtualNetworkGatewayPacketCaptureResult output = new PSVirtualNetworkGatewayPacketCaptureResult()
+            {
+                Name = existingVirtualNetworkGateway.Name,
+                ResourceGroupName = existingVirtualNetworkGateway.ResourceGroupName,
+                Tag = existingVirtualNetworkGateway.Tag,
+                ResourceGuid = existingVirtualNetworkGateway.ResourceGuid,
+                Location = existingVirtualNetworkGateway.Location,
+                
+            };
+            output.StartTime = DateTime.UtcNow;
+            string result = this.VirtualNetworkGatewayClient.StartPacketCapture(this.ResourceGroupName, this.Name, parameters);
```

Comment:
You need to actually implement ShouldProcess here, see the developer documentation for options

Reply:
implemented ShouldProcess for all the commands. Thanks